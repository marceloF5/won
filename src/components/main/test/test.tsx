import { render, screen } from '@testing-library/react'

import Main from '../'

describe('<Main />', () => {
    it('should render Main component', () => {
        const { container } = render(<Main />)

        expect(container.firstChild).toMatchSnapshot()

        expect(
            screen.getByRole('heading', { name: /Advanced React Course/i })
        ).toBeInTheDocument()
    })

    it('should render the colors correctly', () => {
        const { container } = render(<Main />)

        expect(container.firstChild).toHaveStyle({
            'background-color': '#06092b'
        })
    })
})
