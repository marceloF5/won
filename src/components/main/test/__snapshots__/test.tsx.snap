// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render Main component 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  font-size: 2.5rem;
}

.c2 {
  font-size: 2.5rem;
  font-weight: 400;
}

.c3 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    Boilerplate NextJS
  </h1>
  <h2
    class="c2"
  >
    with TypeScript, ReactJS, NextJS and Styled Components
  </h2>
  <img
    alt="A developer in front a screen code in his computer"
    class="c3"
    src="/img/hero-illustration.svg"
  />
</main>
`
